staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -178,11 +178,13 @@ type Histogram struct {
 
 // GetHistogramFromGatherer collects a metric from a gatherer implementing k8s.io/component-base/metrics.Gatherer interface.
 // Used only for testing purposes where we need to gather metrics directly from a running binary (without metrics endpoint).
-func GetHistogramFromGatherer(gatherer metrics.Gatherer, metricName string) (Histogram, error) {
+func GetHistogramFromGatherer(gatherer metrics.Gatherer, metricName string, metricLabelMap map[string]string) (map[string]Histogram, error) {
```

Comment:
@logicalhan I don't understand why we need metricLabelMap instead of just pass the label as a string?

Reply:
I could be misunderstanding this. I thought this method was trying to fetch a metric from the prometheus registry (namely a histogram) and then return the specific metric for a label which which matches some value. If this is completely generic, then we don't know what the concrete labels are going to be, nor will we know which label values we want to target for selection.

Reply:
Both of approaches LGTM, the reason why it's map here is the change in the caller side will be simpler. I will push another PR which just pass a label here as a comparison.

One thing might note here is `HistogramVec` might associate with multiple labels, which means it's better to be a label list for a specific metric and show all those label in the final result, but I cannot see a real needs by far and the process logic will be pretty ugly.